Bugsnag/Client/BugsnagClient.m

```diff
@@ -416,9 +422,7 @@ - (id)initWithConfiguration:(BugsnagConfiguration *)initConfiguration {
                                                          }];
 
         self.breadcrumbs = [[BugsnagBreadcrumbs alloc] initWithConfiguration:self.configuration];
-        if (self.breadcrumbs.cachePath != nil) {
-            bsg_g_bugsnag_data.breadcrumbsPath = strdup(self.breadcrumbs.cachePath.fileSystemRepresentation);
-        }
+        bsg_g_bugsnag_data.breadcrumbs = self.breadcrumbs.context;
```

Comment:
This feels really weird, setting a global from an instance variable... I worry about future refactorings, when deep knowledge of how this works and what's connected to what dims. Is there another way to do this?

Reply:
Agreed, this is not nice, there's a bit of a mismatch here because we have a global `bsg_g_bugsnag_data` but `BugsnagClient` is not strictly a singleton. Note that [a few lines above](https://github.com/bugsnag/bugsnag-cocoa/blob/a6596e573b5979b83c84cd4ab29a5bf5b2da9648/Bugsnag/Client/BugsnagClient.m#L400) we have a similar bit of code:

```objc
bsg_g_bugsnag_data.onCrash = (void (*)(const BSG_KSCrashReportWriter *))self.configuration.onCrashHandler;
```

An alternative approach that could be used is to call from `BSSerializeDataCrashHandler` into a C function implemented in `BugsnagBreadcrumbs.m` - that would at least encapsulate the strangeness.